controlpanel/frontend/static/javascripts/modules/role-names.js

```diff
@@ -0,0 +1,53 @@
+moj.Modules.roleNames = {
+  id: "role_name",
+  selectId: "role_name-select",
+  formClass: "appRoles",
+  roleListEndpoint: "/parameters/form/role-list.js",
+  roles: null,
+
+  init() {
+    if (document.querySelectorAll("." + this.formClass).length) {
+      this.getRoles().done(() => {
+        this.loadRolesToSelect();
+        this.bindEvents();
+        console.log(this.roles);
+      });
+    }
+    this.$repoSlugTypeahead = $('#' + this.id)
+  },
+
+  bindEvents() {
+    $('input[name=app_type]').change(() => {
+      this.loadRolesToSelect();
+    })
+  },
+
+  getRoles() {
+    return $.get(
+      this.roleListEndpoint,
+      (data) => {
+        this.roles = data;
+      }
+    )
+  },
+
+  loadRolesToSelect() {
+    $("#" + this.selectId).find('option:not([value=""])').remove();
+    let roles = [];
+    if (self.roles !== null) {
+      const appType = $('input[name=app_type]:checked').val();
+      if (appType) {
+        roles = this.roles[appType];
+      } else {
+        roles = this.roles['airflow'].concat(this.roles['webapp']);
+      }
```

Comment:
We are transpiling ES6, so if you changed the `parameters-list-roles` view to return a simple list instead of the object with two lists, you could do
```javascript
// this.roles = ['airflow-role-1', 'webapp-role-1', 'airflow-role-2', ...]
let roles = this.roles || [];
if (roles && appType) {
  roles = this.roles.filter(name => name.startsWith(appType));
}
```

Reply:
Have to filter the roles in the api because there are a lot of roles that are not related to this. Also webapp roles are prefixed with `alpha_app` - does the frontend js have access to the environment (dev|alpha)?

Reply:
Yep, had to use regex as webapp start with `(dev|alpha)_app_`